File: output/StarCoder/avatar/Python/Java/codeforces_203_A.java
import java.util.*;
public class codeforces_203_A {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String[] xxs = sc.nextLine().split(" ");
        int x = Integer.parseInt(xxs[0]);
        int t = Integer.parseInt(xxs[1]);
        int a = Integer.parseInt(xxs[2]);
        int b = Integer.parseInt(xxs[3]);
        int da = Integer.parseInt(xxs[4]);
        int db = Integer.parseInt(xxs[5]);
        if (x == 0) {
            System.out.println("YES");
            return;
        }
        int a_time = Math.min(a / da, t - 1);
        int b_time = Math.min(b / db, t - 1);
        for (int i = 0; i <= a_time; i++) {
            for (int j = 0; j <= b_time; j++) {
                if (a - da * i == x || b - db * j == x) {
                    System.out.println("YES");
                    return;
                }
                if (a - da * i + b - db * j == x) {
                    System.out.println("YES");
                    return;
                }
            }
        }
        System.out.println("NO");
    }
}

File: output/StarCoder/avatar/Java/Python/codeforces_203_A.py
import java.io.*;import java.util.*;
public class codeforces_203_A {
	private void solve ( ) {
		int x = nextInt ( ) ;
		int t = nextInt ( ) ;
		int a = nextInt ( ) ;
		int b = nextInt ( ) ;
		int da = nextInt ( ) ;
		int db = nextInt ( ) ;
		int first ;
		int second ;
		boolean ok = false ;
		for ( int i = 0 ; i < t ; i ++ ) {
			first = a - ( da * i ) ;
			for ( int j = 0 ; j < t ; j ++ ) {
				second = b - ( db * j ) ;
				if ( second + first == x || second == x || first == x || x == 0 ) {
					ok = true ;
					break ;
				}
			}
		}
		if ( ok ) out.print ( "YES" ) ;
		else out.print ( "NO" ) ;
	}
	public static void main ( String [ ] args ) {
		new codeforces_203_A ( ).run ( ) ;
	}
	BufferedReader br ;
	StringTokenizer st ;
	PrintWriter out ;
	private void run ( ) {
		try {
			br = new BufferedReader ( new InputStreamReader ( System.in ) ) ;
			out = new PrintWriter ( System.out ) ;
			solve ( ) ;
			br.close ( ) ;
			out.close ( ) ;
		} catch ( IOException e ) {
			e.printStackTrace ( ) ;
		}
	}
	private int nextInt ( ) {
		return Integer.parseInt ( next ( ) ) ;
	}
	private String next ( ) {
		while ( st == null ||! st.hasMoreTokens ( ) ) {
			try {
				st = new StringTokenizer ( br.readLine ( ) ) ;
			} catch ( IOException e ) {
				e.printStackTrace ( ) ;
				return "END_OF_FILE" ;
			}
		}
		return st.nextToken ( ) ;
	}
}